logger/glog/glog.go

```diff
@@ -882,15 +959,45 @@ func (sb *syncBuffer) Write(p []byte) (n int, err error) {
 	return
 }
 
-// rotateFile closes the syncBuffer's file and starts a new one.
-func (sb *syncBuffer) rotateFile(now time.Time) error {
+// shouldRotate checks if we need to rotate the current log file
+func (sb *syncBuffer) shouldRotate(len int, now time.Time) bool {
+	newLen := sb.nbytes + uint64(len)
+	if newLen <= MinSize {
```

Comment:
What about adding

```
if MaxSize == 0 {
	return false
}
```

or?

Reply:
Oh I see, nm. Because we might not care about size, but care just about age. Gotcha.